src/hooks/useBrandConfig.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { StyleSheet } from 'react-native';
+import { TextStyles, defaultTextStyles } from 'src/components/Text';
+import * as theme from '../theme';
+
+export interface BrandConfig {
+  styles: {
+    Text?: TextStyles | undefined;
+  };
+  theme: {};
+}
+
+const BrandConfigContext = React.createContext<BrandConfig>({
+  styles: {
+    Text: defaultTextStyles,
+  },
+  theme,
+});
+
+interface Props {
+  styles?: BrandConfig['styles'];
+  children: React.ReactNode;
+}
+
+export function BrandConfigProvider({ styles, children }: Props) {
```

Comment:
So one can say `<BrandingConfigProvider styles={myStyleOverrides}>...` - very cool.  And eventually `theme` will also be injectable.  And those `styles` and `theme` props can be extensions of the types provided in this lib, with additional props etc.  Then if `useBrandConfig` had a generic, one could get back their own type.  Excited to see all this come together in storybook story/demo in some future PR

Reply:
Correct!  See `TODO:` list at top.  😄